dbt-snowflake/src/dbt/adapters/snowflake/impl.py

```diff
@@ -482,58 +484,37 @@ def build_catalog_relation(self, model: RelationConfig) -> Optional[CatalogRelat
         return None
 
     @available
-    def describe_dynamic_table(self, relation: RelationConfig) -> Dict[str, Any]:
+    def describe_dynamic_table(self, relation: SnowflakeRelation) -> Dict[str, Any]:
         """
         Get all relevant metadata about a dynamic table to return as a dict to Agate Table row
 
         Args:
             relation (SnowflakeRelation): the relation to describe
         """
-
-        original_val: Optional[str] = None
-        try:
-            # Store old QUOTED_IDENTIFIERS_IGNORE_CASE
-            show_param_sql = "show parameters like 'QUOTED_IDENTIFIERS_IGNORE_CASE' in SESSION"
-            show_param_res = self.execute(show_param_sql)
-            if show_param_res:
-                param_qid = show_param_res[0].query_id
-                scan_param_sql = f"SELECT * FROM TABLE(RESULT_SCAN('{param_qid}'))"
-                param_scan_res, rows = self.execute(scan_param_sql, fetch=True)
-                if param_scan_res and param_scan_res.code == "SUCCESS":
-                    try:
-                        original_val = rows[0][1]
-                    except (IndexError, TypeError):
-                        original_val = None
-
-            # falsify QUOTED_IDENTIFIERS_IGNORE_CASE for execution only
-            self.execute("alter session set QUOTED_IDENTIFIERS_IGNORE_CASE = FALSE", fetch=False)
-
-            show_sql = (
-                f"show dynamic tables like '{relation.identifier}' "
-                f"in schema {relation.database}.{relation.schema}"
+        quoting = relation.quote_policy
```

Comment:
is this a reliable single source of truth for what should be quoted where at all times?

How does this address the omission of `QUOTED_IDENTIFIERS_IGNORE_CASE = False` (frankly I'm not sure exactly what it was doing before)